command/meta_providers.go

```diff
@@ -1,11 +1,16 @@
 package command
 
 import (
+	"errors"
 	"fmt"
+	"net"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
+	"strings"
 
+	"github.com/davecgh/go-spew/spew"
```

Comment:
Yeah, sorry, all of this could use a cleanup, I think. Apologies for not clarifying the status; I don't expect much/any logic to change, but I do intend to take another pass at this PR and remove some of the debugging / "just make it work" stuff that's currently there, like this.

I also intend to add some tests :)